src/OpenApi/src/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal static class OpenApiSchemaExtensions
+{
+    /// <summary>
+    /// Generates a deep copy of a given <see cref="OpenApiSchema"/> instance.
+    /// </summary>
+    /// <remarks>
+    /// The copy constructors of the <see cref="OpenApiSchema"/> class do not perform a deep
+    /// copy of the instance which presents a problem whe making modifications in deeply nested
+    /// subschemas. This extension implements a deep copy on <see cref="OpenApiSchema" /> to guarantee
+    /// that modifications on cloned subschemas do not affect the original subschema.
+    /// /// </remarks>
+    /// <param name="schema">The <see cref="OpenApiSchema"/> to generate a deep copy of.</param>
+    public static OpenApiSchema Clone(this OpenApiSchema schema)
+    {
+        return new OpenApiSchema
```

Comment:
Is it worth adding a reflection-based test somewhere that checks for any future version of Microsoft.OpenApi adding new properties to this type and flush out this code needing to be updated?

Reply:
My bad, didn't spot that {homer_backing_away.gif}